lib/chef/azure/core/bootstrap_context.rb

```diff
@@ -21,6 +21,10 @@ def chef_server_ssl_cert
           @chef_config[:chef_server_ssl_cert_content]
         end
 
+        def first_boot
+          first_boot_attributes_and_run_list = (@config[:first_boot_attributes] || {}).merge(:run_list => @run_list)

```

Comment:
You don't use the variable so you can drop the assignment here and just have `(@config[:first_boot_attributes] || {}).merge(:run_list => @run_list)`

A simple version would be `Hash(@config[:first_boot_attributes]).merge(:run_list => @run_list)` as Hash with either return the argument passed as a hash, which it already is, or an empty hash if the argument passed evaluates to nil.